apps/mobile/src/queries/balance/stacks-balance.query.ts

```diff
@@ -0,0 +1,44 @@
+// import { AccountId } from '@/models/domain.model';
+import { QueryObserverResult } from '@tanstack/react-query';
+
+import { Money, StxCryptoAssetBalance } from '@leather.io/models';
+import { useCryptoCurrencyMarketDataMeanAverage } from '@leather.io/query';
+import { baseCurrencyAmountInQuote, createMoney } from '@leather.io/utils';
+
+import { useStxBalancesQueries } from './use-stx-balances.query';
+
+interface StxBalances {
+  totalStxBalance: Money;
+  accounts: QueryObserverResult<StxCryptoAssetBalance, unknown>[];
+}
+
+function useGetStxBalanceByAddresses(addresses: string[]): StxBalances {
+  const { totalData, queries } = useStxBalancesQueries(addresses);
+
+  if (!addresses || addresses.length === 0)
+    return { totalStxBalance: createMoney(0, 'STX'), accounts: [] };
+
+  return {
+    totalStxBalance: totalData,
+    accounts: queries, // TODO as yet unused but perhaps useful
```

Comment:
I'm not sure if I should return the `accounts` here as maybe we will use this data elsewhere so it could be useful to cache if we are getting it for balances anyway